presidio-recognizers-store/cmd/presidio-recognizers-store/main.go

```diff
@@ -0,0 +1,348 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"strconv"
+	"time"
+
+	"google.golang.org/grpc/reflection"
+
+	"flag"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	types "github.com/Microsoft/presidio-genproto/golang"
+
+	"github.com/Microsoft/presidio/pkg/cache"
+	log "github.com/Microsoft/presidio/pkg/logger"
+	"github.com/Microsoft/presidio/pkg/platform"
+	"github.com/Microsoft/presidio/pkg/presidio/services"
+	"github.com/Microsoft/presidio/pkg/rpc"
+)
+
+type server struct{}
+
+var (
+	settings         *platform.Settings
+	recognizersStore cache.Cache
```

Comment:
My view is the Redis is best used as a cache not a persistent storage. While it's easy to use maybe something more focused on persistent data with query support etc would be worth considering. Cosmos can be easily accessed from Golang using the Mongo API, for example. See: https://github.com/lawrencegripper/ion/blob/master/internal/app/handler/dataplane/documentstorage/mongodb/mongodb.go

Reply:
Would that work on-premises as well?

Reply:
In the ION example we used the MongoAPI so this would work with a standard Mongo Server on premise and Cosmos when on Azure. When using more complex Mongo features the Cosmos support is limited but for this usecase I wouldn't see that being a problem.

Reply:
Redis is not the first choice for persistent storage. however it does it well and can be used as an on premise solution as well (containerized).
It can support high scale and fast query time.

Also, the plan is to use this Redis as a cache in other places, so it simplifies the solution to have one such component (while satisfying both performance / requirements)